### PR TITLE
QTable cannot be pickled

### PR DESCRIPTION
**Pull request by zblz**
While pickling an `astropy.table.Table` works fine, QTables are unable to be pickled:

```python
In [3]: from astropy.table import QTable
In [4]: t=QTable()
In [5]: t['a']=np.linspace(1,10)*u.m
In [6]: t['b']=np.linspace(1,10)*u.s
In [7]: pickle.dumps(t)
---------------------------------------------------------------------------
PicklingError                             Traceback (most recent call last)
<ipython-input-7-f90063f6658b> in <module>()
----> 1 pickle.dumps(t)

PicklingError: Can't pickle <class 'weakref'>: attribute lookup weakref on builtins failed
```

This is not a problem in general use, but prevents QTables from being used with multiprocessing, which pickles all the objects it passes to the function calls.

### CODE REVIEW

**Comment by zblz** [comment]
I have added a PR to the original issue with a test for the QTable pickle and a tentative fix: return copies and not views of the columns in the `__getstate__` function of QTable. It works, but I am not sure whether this could cause any problems. @embray, @taldcroft: any ideas on this?

File: astropy/table/table.py [hunk 1]
```diff
@@ -2109,3 +2109,9 @@ def _is_mixin_column(self, col):
         If ``col`` is a string then it refers to a column name in this table.
         """
         return super(QTable, self)._is_mixin_column(col, quantity_is_mixin=True)
+
+    def __getstate__(self):
+        columns_copy = OrderedDict()
+        for key in self.columns.keys():
+            columns_copy[key] = self.columns[key].copy()
```
Comment (embray) [inline astropy/table/table.py]: This can just be written:
```
columns_copy = dict(key, col.copy() for key, col in self.columns.items())
```

**Comment by taldcroft** [comment]
@zblz - thanks for the code! I need to think a little more about your strategy, but it's certainly good news that tests are passing.

In the meantime can you add an explicit test of the unpicked table type (`QTable`) and also type of the original `Quantity`.

It's going a bit beyond the original scope, but I notice now there is no testing of other mixin objects like Time and SkyCoord.  Can you throw in columns for each of those as well?  (But it's OK if you want to stay to the original scope and we can open a different issue).

This will also need a CHANGES.rst update for the 1.0.2 bug fix section.

**Comment by zblz** [comment]
@taldcroft: Good call, I added a test for Time and SkyCoord and ran into further problems. Apparently, SkyCoord objects do not have a ``copy`` function, so it fails on that. I tried putting it into a ``try ... except`` block so it doesn't do a copy for SkyCoord, but then it fails with the weakref error on pickle. 

I have limited the test to Quantity and Time, and will raise an issue to implement a copy function for SkyCoord.

File: astropy/table/table.py [hunk 2]
```diff
@@ -2109,3 +2109,7 @@ def _is_mixin_column(self, col):
         If ``col`` is a string then it refers to a column name in this table.
         """
         return super(QTable, self)._is_mixin_column(col, quantity_is_mixin=True)
+
+    def __getstate__(self):
+        columns = dict((key, col.copy()) for key, col in self.columns.items())
```
Comment (taldcroft) [inline astropy/table/table.py]: Try this:
```
columns = dict((key, col if isinstance(col, BaseColumn) else col_copy(col))
                for key, col in self.columns.items())
```
This *should* do two things (but not tested on my end): (1) not unnecessarily copy regular columns, (2) correctly copy SkyCoord.  `col_copy` is a mixin-safe version of column copying that should always work.

Comment (zblz) [inline astropy/table/table.py]: great, thanks! I'll try that.

**Comment by zblz** [comment]
Using ``col_copy`` everything works fine, and I have squashed the commits to avoid littering the astropy commit log.

**Comment by zblz** [comment]
Two travis jobs fail: one on a network timeout and the other without showing a fail but returning code 254 (?). Could somebody restart these two? Thanks

**Comment by astrofrog** [comment]
@zblz - restarted!

**Comment by taldcroft** [comment]
Thanks @zblz !

**Pull request merged**